[GLIB] WebKit doesn't build with -DENABLE_MALLOC_HEAP_BREAKDOWN=ON

GeorgesStavracas opened the pull request:
#### f5f08704ee110d47da47f6d5cda78145594439f6
<pre>
[GLIB] WebKit doesn&apos;t build with -DENABLE_MALLOC_HEAP_BREAKDOWN=ON
<a href="https://bugs.webkit.org/show_bug.cgi?id=298223">https://bugs.webkit.org/show_bug.cgi?id=298223</a>

Reviewed by Michael Catanzaro.

There were a few issues here:

 * Commit 37d4a6880db95ef5abd introduced a new argument to the
   WTF_DEPRECATED_MAKE_STRUCT_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER()
   macro, but didn&apos;t update all places accordingly

 * There&apos;s a conflicting symbol with the ResourceLoader expansion,
   so rename it.

 * The heap identifier in StyleInterpolationWrapperBase.h was misnamed,
   probably a copy-paste mistake.

 * In FastMalloc.{cpp,h}, relevant parts of the code were compiled
   out due to not having a BENABLE(MALLOC_HEAP_BREAKDOWN) check. Add
   them.

* Source/WebCore/Modules/fetch/FetchLoader.cpp:
* Source/WebCore/Modules/fetch/FetchLoader.h:
* Source/WebCore/Modules/notifications/NotificationResourcesLoader.cpp:
* Source/WebCore/Modules/notifications/NotificationResourcesLoader.h:
* Source/WebCore/css/CSSSelector.h:
* Source/WebCore/css/parser/CSSTokenizer.cpp:
* Source/WebCore/css/parser/CSSTokenizerInputStream.cpp:
* Source/WebCore/fileapi/FileReaderLoader.cpp:
* Source/WebCore/fileapi/FileReaderLoader.h:
* Source/WebCore/loader/LoaderMalloc.cpp:
* Source/WebCore/loader/ResourceLoader.cpp:
* Source/WebCore/loader/ResourceLoader.h:
* Source/WebCore/page/PerformanceEntry.cpp:
* Source/WebCore/style/StyleInterpolation.cpp:
* Source/WebCore/style/StyleInterpolationWrapperBase.h:
* Source/WebCore/style/StyleTreeResolver.h:
* Source/bmalloc/bmalloc/SystemHeap.cpp:
* Source/bmalloc/bmalloc/SystemHeap.h:

Canonical link: <a href="https://commits.webkit.org/299555@main">https://commits.webkit.org/299555@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2148527333f49bc44f774b1e0141b0a29c035f38

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/119041 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/38722 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/29373 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/125248 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/71104 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/120919 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/39418 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/47304 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/90388 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59837 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121994 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/31428 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/106721 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70848 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/30491 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/24837 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/68902 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/111159 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100875 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25023 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/128282 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/117556 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45948 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34721 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/99007 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/46315 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102941 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98786 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25184 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/44245 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/22244 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/42527 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/45818 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/51497 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/146254 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/45285 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37622 "Found 11 new JSC stress test failures: wasm.yaml/wasm/lowExecutableMemory/executable-memory-oom.js.default-wasm, wasm.yaml/wasm/stress/cc-int-to-int-cross-module-with-exception.js.default-wasm, wasm.yaml/wasm/stress/cc-int-to-int-jit-to-llint.js.default-wasm, wasm.yaml/wasm/stress/wasm-js-string-builtins.js.default-wasm, wasm.yaml/wasm/stress/wasm-js-string-builtins.js.wasm-bbq, wasm.yaml/wasm/stress/wasm-js-string-builtins.js.wasm-bbq-no-consts, wasm.yaml/wasm/stress/wasm-js-string-builtins.js.wasm-collect-continuously, wasm.yaml/wasm/stress/wasm-js-string-builtins.js.wasm-eager, wasm.yaml/wasm/stress/wasm-js-string-builtins.js.wasm-eager-jettison, wasm.yaml/wasm/stress/wasm-js-string-builtins.js.wasm-no-cjit ... (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/48630 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46970 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Scony reviewed (commented):
LGTM

mcatanzaro commented on Source/WebCore/style/StyleInterpolationWrapperBase.h:
```diff
@@ -47,7 +47,7 @@ struct Context;
 class WrapperBase {
     WTF_MAKE_NONCOPYABLE(WrapperBase);
-    WTF_DEPRECATED_MAKE_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER(Animation, WrapperBase);
+    WTF_DEPRECATED_MAKE_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER(Animation, Animation);
```
I think this is one big copy/paste error? This isn't Animation.

GeorgesStavracas replied:
It's just matching the `DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(Animation)` identifier. Maybe it wasn't supposed to be called Animation since the beginning, i.e. 15927b4021419ab37c35d7a7deb505db14a8bd20.

mcatanzaro replied:
I suspect so, yes. We should fix it since we're touching it.

I see the fast malloc macros became very confusing since last I checked, and we now have an essay in FastMalloc.h on how to use them properly. I think the arguments are just reversed? The second argument is heap group, so should probably be `WTF_DEPRECATED_MAKE_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER(WrapperBase, Animation)`?

GeorgesStavracas replied:
I've pushed a fix that changed it to `WTF_DEPRECATED_MAKE_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER(WrapperBase, StyleInterpolationWrapperBase)`, since there is another heap identifier named "Animation" already.

GeorgesStavracas committed:
[GLIB] WebKit doesn't build with -DENABLE_MALLOC_HEAP_BREAKDOWN=ON
https://bugs.webkit.org/show_bug.cgi?id=298223

Reviewed by Michael Catanzaro.

There were a few issues here:

 * Commit 37d4a6880db95ef5abd introduced a new argument to the
   WTF_DEPRECATED_MAKE_STRUCT_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER()
   macro, but didn't update all places accordingly

 * There's a conflicting symbol with the ResourceLoader expansion,
   so rename it.

 * The heap identifier in StyleInterpolationWrapperBase.h was misnamed,
   probably a copy-paste mistake.

 * In FastMalloc.{cpp,h}, relevant parts of the code were compiled
   out due to not having a BENABLE(MALLOC_HEAP_BREAKDOWN) check. Add
   them.

* Source/WebCore/Modules/fetch/FetchLoader.cpp:
* Source/WebCore/Modules/fetch/FetchLoader.h:
* Source/WebCore/Modules/notifications/NotificationResourcesLoader.cpp:
* Source/WebCore/Modules/notifications/NotificationResourcesLoader.h:
* Source/WebCore/css/CSSSelector.h:
* Source/WebCore/css/parser/CSSTokenizer.cpp:
* Source/WebCore/css/parser/CSSTokenizerInputStream.cpp:
* Source/WebCore/fileapi/FileReaderLoader.cpp:
* Source/WebCore/fileapi/FileReaderLoader.h:
* Source/WebCore/loader/LoaderMalloc.cpp:
* Source/WebCore/loader/ResourceLoader.cpp:
* Source/WebCore/loader/ResourceLoader.h:
* Source/WebCore/page/PerformanceEntry.cpp:
* Source/WebCore/style/StyleInterpolation.cpp:
* Source/WebCore/style/StyleInterpolationWrapperBase.h:
* Source/WebCore/style/StyleTreeResolver.h:
* Source/bmalloc/bmalloc/SystemHeap.cpp:
* Source/bmalloc/bmalloc/SystemHeap.h:

Canonical link: https://commits.webkit.org/299555@main
